hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/Networking.java

```diff
@@ -63,33 +63,29 @@ void shutdown() {
         flowControlSender.cancel(false);
     }
 
-    void handle(Packet packet) throws IOException {
+    void handle(Packet packet) {
         if (!packet.isFlagRaised(FLAG_JET_FLOW_CONTROL)) {
             handleStreamPacket(packet);
             return;
         }
         handleFlowControlPacket(packet.getConn().getEndPoint(), packet.toByteArray());
     }
 
-    private void handleStreamPacket(Packet packet) throws IOException {
-        BufferObjectDataInput in = createObjectDataInput(nodeEngine, packet.toByteArray());
-        long executionId = in.readLong();
-        int vertexId = in.readInt();
-        int ordinal = in.readInt();
+    private void handleStreamPacket(Packet packet) {
+        MemoryDataInput input = new MemoryDataInput(packet.toByteArray());
```

Comment:
This is one additional allocation, if we are concerned about it we can use `MemoryWriter` directly in trade of readability/maintainability... WDYT?